### PR TITLE
Generate GRPC server

### LINKED ISSUE #2: Generate GRPC server
* Add simple dummy proto definition for TranslateService
* Use buff.build to generate go GRPC server
* Use earthly for tooling - to run buff.build in this case

### PR DESCRIPTION
**Pull request by VladislavsPerkanuks**
Not sure about generated code location

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,3 +13,5 @@
 
 # Dependency directories (remove the comment below to include it)
 # vendor/
+.vscode/
```
Comment (janishorsts) [inline .gitignore]: ignore all except `.vscode/.settings.json` and `.vscode/.launch.json`. We'll share workspace settings for everybody.

File: api/translate/buf.gen.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+version: v1
+managed:
+  enabled: true
+  go_package_prefix:
+    default: "github.com/expect-digital/translate/api/translate/gen/proto/go/translate/translate/v1"
```
Comment (janishorsts) [inline api/translate/buf.gen.yaml]: Output to `github.com/expect-digital/translate/pkg/server/translate/v1`

File: Earthfile [hunk 3]
```diff
@@ -0,0 +1,12 @@
+VERSION 0.6
+
+build:
+    FROM bufbuild/buf
+    COPY api/translate api/translate
+    RUN rm -rf api/translate/gen
+    WORKDIR api/translate
+    RUN buf mod update
+    RUN buf lint
```
Comment (janishorsts) [inline Earthfile]: The `lint` is not part of the `build` target. Please remove. Scope creep.

File: Earthfile [hunk 4]
```diff
@@ -0,0 +1,12 @@
+VERSION 0.6
+
+build:
+    FROM bufbuild/buf
+    COPY api/translate api/translate
+    RUN rm -rf api/translate/gen
```
Comment (janishorsts) [inline Earthfile]: Incorrect use of earthly.

You have to copy only files that are required to complete the target. Generated go files are not part of it. If earthly notices changes in copied files, it deletes the cache and executes the steps again.

File: Earthfile [hunk 5]
```diff
@@ -0,0 +1,12 @@
+VERSION 0.6
+
+build:
+    FROM bufbuild/buf
+    COPY api/translate api/translate
```
Comment (janishorsts) [inline Earthfile]:
```earthfile
COPY --dir api/translate .
WORKDIR translate
RUN buf ...
```

**Review by janishorsts — CHANGES_REQUESTED** [review]

**Comment by janishorsts** [comment]
Can you please add `.env` with the following content for earthly?

```sh
EARTHLY_ALLOW_PRIVILEGED=true
EARTHLY_INSTALLATION_NAME=translate
```

File: api/translate/translate/v1/translate.proto [hunk 6]
```diff
@@ -0,0 +1,6 @@
+syntax = "proto3";
+package translate.v1;
+
```
Comment (klavinsjanis) [inline api/translate/translate/v1/translate.proto:3]: From: https://developers.google.com/protocol-buffers/docs/reference/go-generated
In order to generate Go code, the Go package's import path must be provided for every .proto file (including those transitively depended upon by the .proto files being generated). There are two ways to specify the Go import path:

by declaring it within the .proto file, or
by declaring it on the command line when invoking protoc.
We recommend declaring it within the .proto file so that the Go packages for .proto files can be centrally identified with the .proto files themselves and to simplify the set of flags passed when invoking protoc. If the Go import path for a given .proto file is provided by both the .proto file itself and on the command line, then the latter takes precedence over the former.
The Go import path is locally specified in a .proto file by declaring a go_package option with the full import path of the Go package.

Comment (VladislavsPerkanuks) [reply api/translate/translate/v1/translate.proto:3]: Defined here
https://github.com/expect-digital/translate/blob/fbba7cbcb2094b848695ff61e9823652b1246aa8/api/translate/buf.gen.yaml#L4-L5

https://docs.buf.build/tour/use-managed-mode

**Review by klavinsjanis — APPROVED** [review]

File: Earthfile [hunk 7]
```diff
@@ -0,0 +1,10 @@
+VERSION 0.6
+
+build:
```
Comment (janishorsts) [inline Earthfile]: Please rename target to `proto`. `build` is for different purpose.

File: Earthfile [hunk 8]
```diff
@@ -0,0 +1,10 @@
...
+    FROM bufbuild/buf
```
Comment (janishorsts) [inline Earthfile]: let's use 2 space indentation

File: .gitignore [hunk 9]
```diff
@@ -13,3 +13,6 @@
 
 # Dependency directories (remove the comment below to include it)
 # vendor/
+.vscode/*
+!.vscode/launch.json
+!.vscode/settings.json
```
Comment (janishorsts) [inline .gitignore:18]: missing `pkg/server` in .gitignore

File: Earthfile [hunk 10]
```diff
@@ -0,0 +1,10 @@
...
+    COPY --dir api/translate .
+    WORKDIR translate
+    RUN buf mod update
+    RUN buf build
+    RUN buf generate
```
Comment (janishorsts) [inline Earthfile]: Can you please mount the buf dependencies directory with type=cache? To avoid downloading all dependencies every time we make changes to proto files.

Comment (VladislavsPerkanuks) [reply Earthfile]: Want to discuss about this

Comment (janishorsts) [reply Earthfile]: https://docs.buf.build/ci-cd/setup#ci-caching
https://docs.earthly.dev/docs/guides/advanced-local-caching#option-2-mount-based-caching-advanced

Comment (VladislavsPerkanuks) [reply Earthfile]: Should be okay now

**Review by janishorsts — APPROVED** [review]
